### ISSUE TITLE
Validate value of extension ID before run/install

### ISSUE THREAD

**Issue by kumar303** [mozilla/web-ext#140]
Before the run/install command (https://github.com/mozilla/web-ext/issues/6), the contents of the manifest ID value should be validated. This is because it will be used to write a file to disk in the profile/extensions directory and bad things might happen if the ID is invalid.

**Comment by kumar303** [comment]
possible ID values: https://developer.mozilla.org/en-US/Add-ons/Install_Manifests#id

**Comment by andymckay** [comment]
Is there code in the linter you could re-use for that?

**Comment by eviljeff** [comment]
This would be a good bug for someone who knows Javascript.

Getting started for any contributors: https://wiki.mozilla.org/Add-ons/Contribute/AMO/Code

mentor: @kumar303

**Comment by wagnerand** [comment]
I would add this in `src/util/manifest.jssrc/util/manifest.js:getValidatedManifest` but that would make it run not only during *run*, but also during *build* and *sign*.

@kumar303 is that OK with you?

**Comment by andymckay** [comment]
/me weeps that you wouldn't use the linter or the webextensions-manifest-schema to do all this work for you.

**Comment by wagnerand** [comment]
Is web-ext using any of those during or before *run*? If I understand correctly, we want to catch cases here where developers just wrote their webextension and want to run it locally.

**Comment by andymckay** [comment]
Not right now, but we want to be able to do linting offline in web-ext as per #204, that's one of the main advantages of having a linter in JS. Once we've got that library installed we can call into it and extract metadata and such from the directory and let those libraries take care of all the nitty gritty.

**Comment by wagnerand** [comment]
Ok, so I should hold off on this until #204 is fixed. Thanks!

**Comment by kumar303** [comment]
Yes, we should use the linter (#204) and also I'm not sure if this issue is completely valid due to the changing ID requirements. Just so that there is no confusion from a contributor, I'm going to remove the good first bug label.

**Comment by kumar303** [comment]
FYI: this is low priority because it only applies to [--pre-install](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/web-ext_command_reference#--pre-install) which is mainly for legacy support.
